docs/docs/refs-and-the-dom.md

```diff
@@ -50,9 +50,11 @@ class CustomTextInput extends React.Component {
 }
 ```
 
-React will call the `ref` callback with the DOM element when the component mounts, and call it with `null` when it unmounts.
+React will call the `ref` callback with the DOM element when the component mounts, and call it with `null` when it unmounts. Additionally, if the `ref` callback is defined as an inline function, it will get called twice during each render pass, first with `null` and then again with the DOM element. This is because internally, a new instance of the function is created with each render, so React needs to reset the old instance and set-up the new one. This can be avoided by defining the ref callback as a property on the class, but note that it shouldn't matter in the average `ref` use case.
```

Comment:
I thought it would be a good fit here because it's a place where the ref callback calls are mentioned, and also, for me, the first time I was reading up about refs was also the time I'd appreciate this kind of information. But you're right it's a better fit for the "Caveats" section. I'll update the text later today. Thanks!